File: fyn-flutter-app/lib/features/post/data/models/create_comment_request.dart
class CreateCommentRequest {
  final String content;
  final String? parentCommentId;

  CreateCommentRequest({
    required this.content,
    this.parentCommentId,
  });

  Map<String, dynamic> toJson() {
    return {
      'content': content,
      if (parentCommentId != null) 'parentCommentId': parentCommentId,
    };
  }
}

File: fyn-monolithic/src/main/java/com/fyn_monolithic/dto/request/post/CreateCommentRequest.java
package com.fyn_monolithic.dto.request.post;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Data;

import java.util.UUID;

@Data
public class CreateCommentRequest {

    private UUID parentCommentId;

    @NotBlank
    @Size(max = 1024)
    private String content;
}
